js/graviex.js

```diff
@@ -0,0 +1,1012 @@
+                        'deposit', // Requires access_key, tonce, signature, txid and api secret.
+                        'deposit_address', // Requires access_key, tonce, signature, currency and api secret.
+                        'orders', // Requires access_key, tonce, signature and api secret. Optional market, state, limit, page, order_by
+                        'order', // Requires access_key, tonce, signature and api secret AND id (unique orderid).
+                        'trades/my', // Requires market XXXYYY, access_key, tonce, signature and api secret . Optional: limit, timestamp, from, to, order_by
+                        'trades/history', // Requires market XXXYYY, access_key, tonce, signature and api secret . Optional: limit, timestamp, from, to, order_by
+                        'orders/history', // Requires market XXXYYY, access_key, tonce, signature and api secret . Optional: limit, timestamp, from, to, order_by
+                    ],
+                    'post': [
+                        'orders', // Requires access_key, tonce, signature and api secret AND market, side(sell/buy), volume. Optional price ord_type
+                        'orders/multi', // Requires access_key, tonce, signature and api secret AND market, orders, orders[side(sell/buy)], orders[volume]. Optional orders[price], orders[ord_type]. To create multiple orders at once
+                        'orders/clear', // Requires access_key, tonce, signature and api secret. Optional side(sell/buy) to only clear orders on one side.
+                        'order/delete', // Requires access_key, tonce, signature and api secret AND id (unique orderid).
+                    ],
+                },
+            },
+            'fees': {
+            },
+            'limits': {
+                'amount': {
+                    'min': 0.001,
+                    'max': undefined,
+                },
+            },
+            'precision': {
+                'amount': 8,
+                'price': 8,
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets ();
+        // {
+        //     "id": "giobtc",
+        //     "name": "GIO/BTC",
+        // };
+        const markets = response;
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = this.safeString (market, 'id');
+            let symbolParts = this.safeString (market, 'name');
+            symbolParts = symbolParts.split ('/');
+            const baseId = symbolParts[0];
+            const quoteId = symbolParts[1];
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const active = true;
+            result.push ({
+                'id': id,
```

Comment:
There are a lot of missing fields here that should be filled in as true (spot for instance) false (future, option,swap, ...), or undefined (limits, fees,...).

Please use this as a reference https://github.com/ccxt/ccxt/blob/master/ccxt.d.ts#L55